packages/studio-base/src/players/RosbridgePlayer.ts

```diff
@@ -476,14 +477,129 @@ export default class RosbridgePlayer implements Player {
       }
 
       const problemId = `message:${topicName}`;
-      topic.subscribe((message) => {
+
+      this.#rosClient.addTopicCallback(topicName, (message) => {
         if (!this.#providerTopics) {
           return;
         }
         try {
           const buffer = (message as { bytes: ArrayBuffer }).bytes;
           const bytes = new Uint8Array(buffer);
-          const innerMessage = messageReader.readMessage(bytes);
+          // const innerMessage = messageReader.readMessage(bytes);
+          if ( message._topic_type === 'sensor_msgs/msg/LaserScan' ) {
+            let rbounds = message._ranges_uint16.bounds;
+
+            let rdata = this._base64decode(message._ranges_uint16.points);
+
+            let rview = new DataView(rdata);
+            let num_ranges = rdata.byteLength / 2;
```

Comment:
why / 2 ?

Reply:
We are transforming an array with data of 16 bits into an array of float 32 bits, so `num_ranges` there is used as the length of the 32 bits data array.